# Import Library

In [2]:
import pandas as pd
from ucimlrepo import fetch_ucirepo 

# Datasets Understanding

## Data Loading

In [3]:

# fetch dataset 
phishing_websites = fetch_ucirepo(id=327) 
  
# data (as pandas dataframes) 
X = phishing_websites.data.features 
y = phishing_websites.data.targets 
  
# metadata 
print(phishing_websites.metadata) 
  
# variable information 
print(phishing_websites.variables) 


{'uci_id': 327, 'name': 'Phishing Websites', 'repository_url': 'https://archive.ics.uci.edu/dataset/327/phishing+websites', 'data_url': 'https://archive.ics.uci.edu/static/public/327/data.csv', 'abstract': 'This dataset collected mainly from: PhishTank archive, MillerSmiles archive, Google’s searching operators.', 'area': 'Computer Science', 'tasks': ['Classification'], 'characteristics': ['Tabular'], 'num_instances': 11055, 'num_features': 30, 'feature_types': ['Integer'], 'demographics': [], 'target_col': ['result'], 'index_col': None, 'has_missing_values': 'no', 'missing_values_symbol': None, 'year_of_dataset_creation': 2012, 'last_updated': 'Tue Mar 05 2024', 'dataset_doi': '10.24432/C51W2X', 'creators': ['Rami Mohammad', 'Lee McCluskey'], 'intro_paper': {'ID': 396, 'type': 'NATIVE', 'title': 'An assessment of features related to phishing websites using an automated technique', 'authors': 'R. Mohammad, F. Thabtah, L. Mccluskey', 'venue': 'International Conference for Internet Tec

In [12]:
X['having_ip_address'].unique()

array([-1,  1])

In [ ]:
boolean_col = []
category_col = []

for col in X.columns:
    if X[col].nunique() == 2:
        boolean_col.append(col)
    else:
        category_col.append(col)




In [15]:
X[boolean_col]

having_ip_address  shortining_service  having_at_symbol  \
0                     -1                   1                 1   
1                      1                   1                 1   
2                      1                   1                 1   
3                      1                   1                 1   
4                      1                  -1                 1   
...                  ...                 ...               ...   
11050                  1                   1                -1   
11051                 -1                   1                -1   
11052                  1                   1                 1   
11053                 -1                   1                 1   
11054                 -1                   1                 1   

       double_slash_redirecting  prefix_suffix  domain_registration_length  \
0                            -1             -1                          -1   
1                             1             -1                          -1   
2                             1             -1                          -1   
3                             1             -1                           1   
4                             1             -1                          -1   
...                         ...            ...                         ...   
11050                         1              1                          -1   
11051                        -1             -1                          -1   
11052                         1             -1                          -1   
11053                         1             -1                           1   
11054                         1             -1                           1   

       favicon  port  https_token  request_url  ...  redirect  on_mouseover  \
0            1     1           -1            1  ...         0             1   
1            1     1           -1            1  ...         0             1   
2            1     1           -1            1  ...         0             1   
3            1     1           -1           -1  ...         0             1   
4            1     1            1            1  ...         0            -1   
...        ...   ...          ...          ...  ...       ...           ...   
11050       -1    -1            1            1  ...         0            -1   
11051       -1    -1            1            1  ...         1            -1   
11052        1     1            1            1  ...         0             1   
11053       -1     1            1           -1  ...         0            -1   
11054        1     1            1           -1  ...         0             1   

       rightclick  popupwindow  iframe  age_of_domain  dnsrecord  page_rank  \
0               1            1       1             -1         -1         -1   
1               1            1       1             -1         -1         -1   
2               1            1       1              1         -1         -1   
3               1            1       1             -1         -1         -1   
4               1           -1       1             -1         -1         -1   
...           ...          ...     ...            ...        ...        ...   
11050          -1           -1      -1              1          1         -1   
11051           1           -1       1              1          1          1   
11052           1            1       1              1          1         -1   
11053           1           -1       1              1          1         -1   
11054           1            1       1             -1          1         -1   

       google_index  statistical_report  
0                 1                  -1  
1                 1                   1  
2                 1                  -1  
3                 1                   1  
4                 1                   1  
...             ...                 ...  
11050             1                   1  
11051             1                   1  
1

In [17]:
X[category_col]

url_length  having_sub_domain  sslfinal_state  url_of_anchor  \
0               1                 -1              -1             -1   
1               1                  0               1              0   
2               0                 -1              -1              0   
3               0                 -1              -1              0   
4               0                  1               1              0   
...           ...                ...             ...            ...   
11050          -1                  1               1              1   
11051           1                  1              -1             -1   
11052          -1                  1              -1              0   
11053          -1                 -1              -1             -1   
11054          -1                 -1              -1             -1   

       links_in_tags  sfh  web_traffic  links_pointing_to_page  
0                  1   -1           -1                       1  
1                 -1   -1            0                       1  
2                 -1   -1            1                       0  
3                  0   -1            1                      -1  
4                  0   -1            0                       1  
...              ...  ...          ...                     ...  
11050              1   -1           -1                       1  
11051             -1    0            1                      -1  
11052             -1   -1            1                       0  
11053              1   -1            1                       1  
11054              0   -1           -1                       1  

[11055 rows x 8 columns]

Ada 11055 Baris data dalam datasets tersebut, dengan jumlah fitur sebanyak 30 

Dari 30 Fitur, 22 fitur merupakan fitur tipe boolean (True or False) dan 8 sisanya merupakan fitur kategori (netral,positif,negatif)

## Analysis Variable

In [18]:
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11055 entries, 0 to 11054
Data columns (total 30 columns):
 #   Column                      Non-Null Count  Dtype
---  ------                      --------------  -----
 0   having_ip_address           11055 non-null  int64
 1   url_length                  11055 non-null  int64
 2   shortining_service          11055 non-null  int64
 3   having_at_symbol            11055 non-null  int64
 4   double_slash_redirecting    11055 non-null  int64
 5   prefix_suffix               11055 non-null  int64
 6   having_sub_domain           11055 non-null  int64
 7   sslfinal_state              11055 non-null  int64
 8   domain_registration_length  11055 non-null  int64
 9   favicon                     11055 non-null  int64
 10  port                        11055 non-null  int64
 11  https_token                 11055 non-null  int64
 12  request_url                 11055 non-null  int64
 13  url_of_anchor               11055 non-null  int64
 14  links_

Data sudah berbentuk numerik dengan dua macam tipe

#### Data tipe boolean
Data dengan tipe ini terdapat sejumlah 22 kolom, data ini merepresentasikan sebuah kondisi, contohnya kolom `having_ip_address` memiliki nilai 1 dan -1, yang artinya 1 jika link tersebut memiliki ip address dan -1 jika tidak memiliki ip address

#### Data tipe category
Terdapat 8 kolom bertipe kategori yang merepresentasikan tingkat kondisi suatu fitur. Contohnya, kolom URL_Length memiliki nilai:

1 → aman (URL pendek),

0 → mencurigakan (panjang sedang),

-1 → phishing (URL sangat panjang).

Nilai-nilai ini bersifat ordinal, bukan numerik murni, dan menunjukkan level risiko. Kolom lain yang serupa antara lain Request_URL, Subdomain_Level, dan HTTPS_Token.
